### ISSUE TITLE
Support strings for frame in HEALPix class

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy-healpix#56]
We should try and re-use the frame-parsing machinery from Astropy to make it so we can use:

```
hp = HEALPix(nside=4, order='ring', frame='galactic')
```

(like in SkyCoord)

**Comment by cdeil** [comment]
👍 

This is one thing I'm missing in https://github.com/hipspy/hips/pull/109 , trying to use `astropy-healpix` for `hips`. I'll implement some utility function there now, but having direct support for `frame={‘icrs’, ‘galactic’, ‘ecliptic’}`would be great (I haven't encountered ecliptic so far, but the other two are common).

**Comment by astrofrog** [comment]
This should be easy to do with:

```
In [1]: from astropy.coordinates import frame_transform_graph

In [3]: frame_transform_graph.lookup_name('galactic')
Out[3]: astropy.coordinates.builtin_frames.galactic.Galactic
```

**Comment by astrofrog** [comment]
(This could replace ``make_frame``)

**Comment by cdeil** [comment]
I don't know ... this silently fails (returns `None`):
```
frame_transform_graph.lookup_name('spam')
```
For now I'll stick with my helper function.

**Comment by astrofrog** [comment]
We could easily check for ``None`` output though and raise an exception in that case? (for here, not hips). I'll open an issue to Astropy to add an option to raise an exception if the frame is not known.

**Comment by cdeil** [comment]
Yes, maybe that would be the way to go here.

I think the hard part will be to have a sane way to handle frame for FITS I/O for HEALPix maps. There only a small subset of frames will be supported, so at that point something custom like the helper function I have now in `hipspy` or what you have in `reproject` is needed.

I guess a good strategy is to allow `HEALPix` objects with any `frame` for in-memory computations, and only to apply the checks on valid subset on FITS I/O?

**Comment by astrofrog** [comment]
> I guess a good strategy is to allow HEALPix objects with any frame for in-memory computations, and only to apply the checks on valid subset on FITS I/O?

I agree with this! The HEALPix class itself should not have restrictions, whereas once we write the FITS readers/writers, those should have validation of ``COORDSYS``.
